File: backend/src/main/java/com/java/NBE4_5_3_7/standard/Ut.java
package com.java.NBE4_5_3_7.standard;

import com.java.NBE4_5_3_7.global.app.AppConfig;
import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.ObjectMapper;
import io.jsonwebtoken.ExpiredJwtException;
import io.jsonwebtoken.Jwts;
import io.jsonwebtoken.security.Keys;

import javax.crypto.SecretKey;
import java.util.Date;
import java.util.Map;

public class Ut {
    public static class Json {
        private static final ObjectMapper objectMapper = AppConfig.getObjectMapper();

        public static String toString(Object obj) {
            try {
                return objectMapper.writeValueAsString(obj);
            } catch (JsonProcessingException e) {
                throw new RuntimeException(e);
            }
        }
    }

    public static class Jwt {
        public static String createToken(String keyString, int expireSeconds, Map<String, Object> claims) {
            SecretKey secretKey = Keys.hmacShaKeyFor(keyString.getBytes());
            Date issuedAt = new Date();
            Date expiration = new Date(issuedAt.getTime() + 1000L * expireSeconds);
            return Jwts.builder()
                    .claims(claims)
                    .issuedAt(issuedAt)
                    .expiration(expiration)
                    .signWith(secretKey)
                    .compact();
        }

        public static boolean isValidToken(String keyString, String token) {
            try {
                SecretKey secretKey = Keys.hmacShaKeyFor(keyString.getBytes());
                Jwts.parser().verifyWith(secretKey).build().parse(token);
            } catch (Exception e) {
                e.printStackTrace();
                return false;
            }
            return true;
        }

        @SuppressWarnings("unchecked")
        public static Map<String, Object> getPayload(String keyString, String jwtStr) {
            SecretKey secretKey = Keys.hmacShaKeyFor(keyString.getBytes());
            try {
                return (Map<String, Object>) Jwts.parser()
                        .verifyWith(secretKey)
                        .build()
                        .parse(jwtStr)
                        .getPayload();
            } catch (ExpiredJwtException e) {
                return null;
            } catch (Exception e) {
                e.printStackTrace();
                return null;
            }
        }
    }
}

File: backend/src/main/java/com/java/NBE4_5_3_7/standard/Ut.kt
package com.java.NBE4_5_3_7.standard;

import io.jsonwebtoken.Jwts
import io.jsonwebtoken.security.Keys
import java.util.*
import javax.crypto.SecretKey

class Ut {

    object Jwt {

        @JvmStatic
        fun createToken(keyString: String, expireSeconds: Int, claims: Map<String, Any>): String {
            val secretKey: SecretKey = Keys.hmacShaKeyFor(keyString.toByteArray())
            val issuedAt = Date()
            val expiration = Date(issuedAt.time + 1000L * expireSeconds)

            return Jwts.builder()
                .claims(claims)
                .issuedAt(issuedAt)
                .expiration(expiration)
                .signWith(secretKey)
                .compact()
        }

        @JvmStatic
        fun isValidToken(keyString: String, token: String?): Boolean {
            try {
                val secretKey = Keys.hmacShaKeyFor(keyString.toByteArray())

                Jwts
                    .parser()
                    .verifyWith(secretKey)
                    .build()
                    .parse(token)
            } catch (e: Exception) {
                e.printStackTrace()
                return false
            }

            return true
        }

        @JvmStatic
        fun getPayload(keyString: String, jwtStr: String?): Map<String, Any> {
            val secretKey = Keys.hmacShaKeyFor(keyString.toByteArray())

            return Jwts
                .parser()
                .verifyWith(secretKey)
                .build()
                .parse(jwtStr)
                .payload as Map<String, Any>
        }
    }
}
